node/actors/network/src/gossip/mod.rs

```diff
@@ -148,37 +161,91 @@ impl Network {
+            loop {
+                let quorum_opt = {
+                    let votes = sync::changed(ctx, &mut recv_votes).await?;
+                    votes.find_quorum(attesters, &genesis)
+                };
+
+                if let Some(qc) = quorum_opt {
+                    self.batch_store
+                        .persist_batch_qc(ctx, qc)
+                        .await
+                        .wrap("persist_batch_qc")?;
+
+                    continue 'status;
+                }
+            }
+        }
+    }
+
+    /// Poll the attestation status and update the watch.
+    pub(crate) async fn run_attestation_client(&self, ctx: &ctx::Ctx) -> ctx::Result<()> {
+        if self.genesis().attesters.is_none() {
+            tracing::info!("no attesters in genesis, not polling the attestation status");
+            return Ok(());
+        };
+
+        const POLL_INTERVAL: Duration = Duration::seconds(5);
+
         loop {
-            // In the future when we might be gossiping about multiple batches at the same time,
-            // we can collect the ones we submitted into a skip list until we see them confirmed
-            // on L1 and we can finally increase the minimum as well.
-            let quorums = {
-                let votes = sync::changed(ctx, &mut sub).await?;
-                votes.find_quorums(attesters, &genesis, |_| false)
-            };
-
-            for qc in quorums {
-                // In the future this should come from confirmations, but for now it's best effort, so we can forget ASAP.
-                // TODO: An initial value could be looked up in the database even now.
-                let next_batch_number = qc.message.number.next();
-
-                self.batch_store
-                    .persist_batch_qc(ctx, qc)
-                    .await
-                    .wrap("persist_batch_qc")?;
-
-                self.batch_votes
-                    .set_min_batch_number(next_batch_number)
-                    .await;
+            match self
+                .attestation_status_client
+                .next_batch_to_attest(ctx)
+                .await
+            {
+                Ok(Some(batch_number)) => {
+                    self.attestation_status.update(batch_number).await;
+                    // We could also update the minimum batch number here, which might
+                    // help mitigate the problem of missing a vote if the batch number
+                    // happened to decrease. But we decided to fix it at the source,
+                    // so the only place that is adjusted is before looking for a QC.
+                }
+                Ok(None) => tracing::debug!("waiting for attestation status..."),
+                Err(error) => tracing::error!(
+                    ?error,
+                    "failed to poll attestation status, retrying later..."
+                ),
             }
+            ctx.sleep(POLL_INTERVAL).await?;
```

Comment:
that's the sleep that I would like to avoid in era-consensus

Reply:
I don't fancy hardcoded polls either, just following the pattern. They might have been originally more prevalent in `zksync-era`, and I started adding them to places like the `AttestationRunner`, introducing them to `era-consensus. Maybe we could just make them a parameter, so at least they are easy to change from `zksync-era`? Although getting them to where they need to be might be awkward, with miscellaneous poll interval defaults being added to the `Config` of the `Executor.

Reply:
so the patter that I was trying to maintain so far was to have the polls in zksync-era, where they are just unavoidable because you cannot subscribe to postgres db afaik.